vw/src/main/scala/com/eharmony/spotz/objective/vw/VwFunctions.scala

```diff
@@ -0,0 +1,157 @@
+package com.eharmony.spotz.objective.vw
+
+import com.eharmony.spotz.Preamble.Point
+
+import scala.collection.mutable
+
+trait VwFunctions {
+
+  /**
+    * Merge hyperparameters from the point object into VW parameter map.  Hyperparameter
+    * values will overwrite a map value during the merge if a hyerparameter label is the
+    * same as a map key.
+    *
+    * Certain VW parameters are not allowed to be user specified because the objective
+    * function needs control over them, ie. parameters related to cache, dataset,
+    * testing, final regressor, etc.  These parameters are removed from the map so that
+    * the objective function can specify them manually.
+    *
+    * @param vwParamMap a Map[String, String] where the key is a VW argument and the value
+    *                   is the argument value.
+    * @param point a point object representing the hyper parameter values
+    * @return a new Map[String, String] which is the result of merging the vwParamMap and
+    *         the point.
+    */
+  def mergeVwParams(vwParamMap: Map[String, _], point: Point): Map[String, _] = {
+    val vwParamsMutableMap = mutable.Map[String, Any](vwParamMap.toSeq: _*)
+
+    point.getHyperParameterLabels.foldLeft(vwParamsMutableMap) { (mutableMap, vwHyperParam) =>
+      if (mutableMap.contains(vwHyperParam)) {
+        val listValues = mutableMap(vwHyperParam) match {
+          case it: Iterable[_] => it.toList ++ point.get(vwHyperParam)
+          case value => List(value, point.get(vwHyperParam))
+        }
+        mutableMap += ((vwHyperParam, listValues))
+      } else {
+        mutableMap += ((vwHyperParam, point.get(vwHyperParam)))
+      }
+
+      mutableMap
+    }
+
+    removeParams(vwParamsMutableMap).toMap
```

Comment:
It's not, toMap converts it to immutable.  I keep it mutable throughout the processing but ensure that it becomes immutable outside this code.  However, I will make these methods private as I don't feel they should be accessible outside of a few entry points.